src/mesh/mod.rs

```diff
@@ -114,23 +110,53 @@ impl Mesh {
                                 &vertices_collection[cast_usize(v3)],
                             );
 
-                            normals_collection.push(face_normal);
-
+                            normals_collection_sharp.push(face_normal);
                         }
-                        // FIXME: once we add other kinds of faces, they must panic here
-                        // _ => panic!("Face must be a triangle, we just created it"),
```

Comment:
Let's not remove this fixme. It is still valid. We are in a constructor that creates a mesh from triangles - any other kinds of faces are really programmer error here.

Reply:
ok